### ISSUE TITLE
Auto annotation using own data set 

### ISSUE THREAD

**Issue by SabinaUS** [openvinotoolkit/cvat#3964]
Hello, I am a little confused with serverless tutorial, not only me in this situation https://stackoverflow.com/questions/69720376/how-to-use-yolo-v4-tiny-trained-model-for-auto-annotation-in-cvat. Can you create a video instruction of steps, or here someone that have successful upload own dataset and used auto annotation. Can you share with your experience

**Comment by AmitN1212** [comment]
> Hello, I am a little confused with serverless tutorial, not only me in this situation https://stackoverflow.com/questions/69720376/how-to-use-yolo-v4-tiny-trained-model-for-auto-annotation-in-cvat. Can you create a video instruction of steps, or here someone that have successful upload own dataset and used auto annotation. Can you share with your experience

I deployed a few models and used auto annotation with them. What was confusing about the instructions? I can try to help you...

**Comment by SabinaUS** [comment]
Hello  @AmitN1212  thank you for your kind help. I started with adding as said in tutorial with retinanet model and I think I added it, but What I have to do next to access my added model?

NAMESPACE |                      NAME                      | PROJECT | STATE | NODE PORT | REPLICAS  
  nuclio    | pth-foolwood-siammask                          | cvat    | ready |     49153 | 1/1       
  nuclio    | pth.facebookresearch.detectron2.retinanet_r101 | cvat    | ready |     49154 | 1/1

**Comment by SabinaUS** [comment]
Okay I opened nuclio dashboard at localhost:8070 and now how can I use my model to auto annotation. What the next step
 @AmitN1212

**Comment by SabinaUS** [comment]
![Screenshot from 2021-12-01 13-27-27](https://user-images.githubusercontent.com/61276853/144172893-30c34100-78a8-4d02-b3ae-1147fb98827a.png)
I come until here. What should do next 
Thank you

**Comment by AmitN1212** [comment]
> Okay I opened nuclio dashboard at localhost:8070 and now how can I use my model to auto annotation. What the next step @AmitN1212

It seems that you have deployed the models successfully. 
check [this](https://openvinotoolkit.github.io/cvat/docs/manual/advanced/automatic-annotation/) out. Don't hesitate to ask me again if you don't succeed to use it.

**Comment by SabinaUS** [comment]
@AmitN1212 Thank you so much. Now I am not installed cvat, but using it by cvat.org. And after deploying the model I count not find model in the list of models. Maybe I am wrong, after deploying the model I have to find it in the list of models or some another place. Thank you for your time

**Comment by AmitN1212** [comment]
> @AmitN1212 Thank you so much. Now I am not installed cvat, but using it by cvat.org. And after deploying the model I count not find model in the list of models. Maybe I am wrong, after deploying the model I have to find it in the list of models or some another place. Thank you for your time

When you use CVAT.org, you connect to a remote host, that uses its own pre-installed models. I think you should [setup cvat](https://openvinotoolkit.github.io/cvat/docs/administration/basics/installation/#ubuntu-1804-x86_64amd64) on your own, in your local host.

**Comment by SabinaUS** [comment]
@AmitN1212  Thank you so much

**Comment by AmitN1212** [comment]
You are welcome. You can ask me for help if you face some problems during the installation process.
Have a great day!

**Comment by AmitN1212** [comment]
> @AmitN1212 Thank you so much

You are welcome. You can ask me for help if you face some problems during the installation process.
Have a great day!

**Comment by SabinaUS** [comment]
Thank you so much @AmitN1212 In this case I will not close this issue yet. Thank you I will back again, because I am very new in the using cvat. Thank you so much

**Comment by SabinaUS** [comment]
Hello @AmitN1212. I started work with my own model including to cvat. Before I successfully included retinanet using tutorial. And now I have a problem, if I want to add my own model what should I write on above part of function.yml file. Thank you
directives:
      preCopy:
        - kind: ENV
          value: DEBIAN_FRONTEND=noninteractive
        - kind: RUN
          value: apt-get update && apt-get -y install curl git python3 python3-pip
        - kind: WORKDIR
          value: /opt/nuclio
        - kind: RUN
          value: pip3 install torch==1.8.1+cpu torchvision==0.9.1+cpu torchaudio==0.8.1 -f https://download.pytorch.org/whl/torch_stable.html
        - kind: RUN
          value: pip3 install 'git+https://github.com/facebookresearch/detectron2@v0.4'
        - kind: RUN
          value: curl -O https://dl.fbaipublicfiles.com/detectron2/COCO-Detection/retinanet_R_101_FPN_3x/190397697/model_final_971ab9.pkl
        - kind: RUN
          value: ln -s /usr/bin/pip3 /usr/local/bin/pip

**Comment by AmitN1212** [comment]
> what should I write on below part of function.yml file
In the directives, you should install the dependencies that your code (model) needs to run successfully.

For example,  one of retinanet's dependencies is pytorch, so here:


> value: pip3 install torch==1.8.1+cpu torchvision==0.9.1+cpu torchaudio==0.8.1 -f https://download.pytorch.org/whl/torch_stable.html

Pytorch is being installed. 

Just try to replace the commands here to commands that will install the dependencies of your model.

**Comment by SabinaUS** [comment]
Thank you so much  @AmitN1212. Now I want to include yolact model, it is model for instance segmentation. For making main.py I have to change rectangle to polygon and try to get point list? Can you give me some advice for segmentation with main.py I have to take as example. Thank you for your time
for box, score, label in zip(pred_boxes, scores, pred_classes):
        label = COCO_CATEGORIES[int(label)]["name"]
        if score >= threshold:
            results.append({
                "confidence": str(float(score)),
                "label": label,
                "points": box.tolist(),
                "type": "rectangle",
            })

**Comment by AmitN1212** [comment]
Sorry, I deployed my own models only for object detection with rectangles, so I don't know exactly how to do what you are asking for. However, I assume that you can do whatever you want as far as you send the data back to CVAT in the same 
 required format. First of all, I would suggest you to deploy the code with returning an empty list, just for making sure that all the dependencies have been installed well.

**Comment by SabinaUS** [comment]
Okay Thank you so much  @AmitN1212. Thank you for your kind help

**Comment by AdimasBarelang** [comment]
> > Hello, I am a little confused with serverless tutorial, not only me in this situation https://stackoverflow.com/questions/69720376/how-to-use-yolo-v4-tiny-trained-model-for-auto-annotation-in-cvat. Can you create a video instruction of steps, or here someone that have successful upload own dataset and used auto annotation. Can you share with your experience
> 
> I deployed a few models and used auto annotation with them. What was confusing about the instructions? I can try to help you...

can u share how to do it ? i dont really understand with the tutorial. I want use darknet to auto-annotation in CVAT.
